### PR TITLE
Updates the Raft storage demo scripting for local operations

### PR DESCRIPTION
**Pull request by burtlo**
- Merges `network.sh` and `raft-cluster-demo.sh`

> When moving through the guide quickly I missed this step.

- Creates a setup, start, stop, and clean parameter so the script can be executed in different modes

> I found that if I made a mistake or I had something missing (like the local loopbacks missing) I was often left in a bad state. The guide didn't have any instructions on how to remove the local loopbacks. The `pkill` command in the guide would stop all vault processes and not just the ones launched by this guide.

- Adds better logging of the operations

> The output was not very understandable and it was easy to lose focus on what was being accomplished by the script. It was hard to find the root token for vault 1 in the output

- Adds a status command

> This shows the status of all the vault servers more easily. Though this is not used in the guide.

### CODE REVIEW

**Comment by burtlo** [comment]
I haven't added the recovery mode related content but I don't think that there will be a need to modify this script.

**Comment by yhyakuna** [comment]
Because the loopback address commands only work for Mac or Linux and also, it needs to be executed once, I prefer that portion to be still **_separate_** from the Vault setup.  (I initially thought about adding a reference to Windows doc, but it was too involved, so I left it up to the Windows users to figure it out themselves.)

Don't forget to update the README file as well.

**Comment by burtlo** [comment]
@yhyakuna I assume that the majority of the commands don't work on Windows. I guess more of them would work with Powershell and some of its aliases.

I could move the network operations to another argument and keep it in the same file `./raft-storage-demo.sh setup-network`?

**Comment by yhyakuna** [comment]
No, that's okay.  Let's just go with this.  You're right, pretty much everything is written _Windows users unfriendly_.  

Keep the script as-is, but please don't forget to update the README.

**Comment by burtlo** [comment]
Redefines the local HA cluster experience.

 - Updates README to show the latest usage

 -  Changes `local-test` to `local`

  This is for setting up things locally and it the suffix did not feel like it
  added much value to the name.

- Changes `raft-cluster-demo.sh` to `cluster.sh`

  The name here seemed long and the context felt understood. When I thought
  about the tooling it felt like it was all about managing the cluster and
  not about a single script at this point.

- The `./cluster.sh` command supports commands

  This change creates a more modular approach to the way to do the creation and
  setting up of resources. There are also some helper methods to get clean and
  get the status of all resources. Finally, it provides alias commands to the
  various vault resources in this cluster to make it possible for the learner
  to execute all operations in the same terminal reducing complexity.

File: operations/raft-storage/local/cluster.sh [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+  echo $RECOVERY_KEY2 > recovery_key-vault_2
+  echo $VAULT_TOKEN2 > root_token-vault_2
+
+  printf "\n%s" \
+    "[vault_2] Recovery key: $RECOVERY_KEY2" \
+    "[vault_2] Root token: $VAULT_TOKEN2" \
+    ""
+
+  printf "\n%s" \
+    "[vault_2] waiting to join Vault cluster (15 seconds)" \
```
Comment (yhyakuna) [inline operations/raft-storage/local/cluster.sh]: What does this for?  I believe you can delete this???

File: operations/raft-storage/local/cluster.sh [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+
+  for vault_log in $DEMO_HOME/vault_1.log $DEMO_HOME/vault_2.log $DEMO_HOME/vault_3.log $DEMO_HOME/vault_4.log ; do
+    if [[ -f "$vault_log" ]] ; then
+      printf "\n%s" \
+        "Removing log file $vault_log"
+
+      rm $vault_log
+    fi
+  done
+
```
Comment (yhyakuna) [inline operations/raft-storage/local/cluster.sh]: Remove the `root_token-vault_1` and `root_token-vault_2` as well for the completeness.  

Anyway, this `clean` command is nice!

**Comment by yhyakuna** [comment]
Other than those minor updates that I've made, it all looks good.  

Let's wait for your guide updates and merge the 2 PRs at the same time.

**Pull request merged**